### ISSUE TITLE
Upgraded to 2.1.3 and unable to update and save customers

### ISSUE THREAD

**Issue by endergreen** [magento/magento2#8284]
### Preconditions

1. Upgrade from 2.1.0 to 2.1.3
2. PHP version 7.0.14
3. MySQL version 5.6.34

### Steps to reproduce

1. Log into Admin
2. Bring up the Customer Grid
3. Do an inline edit of the customer and change the group
4. Click the same button

### Expected result

1. Customer saved successfully with new group association

### Actual result

1. The red alert message displays on top of the grid saying: 

[Customer ID: 4] We can't save the customer.

In the exception log the following stack trace complaining about not being able to delete from the customer_entity table because of unknown column value_id in where clause.

[2017-01-25 19:04:32] main.CRITICAL: PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'value_id' in 'where clause' in /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php:228
Stack trace:
#0 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php(228): PDOStatement->execute(Array)
#1 /xxx/xxx/xxx/vendor/magento/framework/DB/Statement/Pdo/Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#2 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Statement.php(303): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#3 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#4 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('DELETE FROM `cu...', Array)
#5 /xxx/xxx/xxx/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(465): Zend_Db_Adapter_Pdo_Abstract->query('DELETE FROM `cu...', Array)
#6 /xxx/xxx/xxx/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(520): Magento\Framework\DB\Adapter\Pdo\Mysql->_query('DELETE FROM `cu...', Array)
#7 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(664): Magento\Framework\DB\Adapter\Pdo\Mysql->query('DELETE FROM `cu...')
#8 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/AbstractEntity.php(1537): Zend_Db_Adapter_Abstract->delete('customer_entity', '(value_id IN ('...')
#9 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/AbstractEntity.php(1455): Magento\Eav\Model\Entity\AbstractEntity->_processAttributeValues()
#10 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/VersionControl/AbstractEntity.php(91): Magento\Eav\Model\Entity\AbstractEntity->_processSaveData(Array)
#11 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Eav\Model\Entity\VersionControl\AbstractEntity->save(Object(Magento\Customer\Model\Backend\Customer))
#12 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Model\ResourceModel\Customer\Interceptor->___callParent('save', Array)
#13 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\Customer\Interceptor), Array, 'clean_cache')
#14 /xxx/xxx/xxx/vendor/magento/framework/App/Cache/FlushCacheByTags.php(71): Magento\Customer\Model\ResourceModel\Customer\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Customer\Model\Backend\Customer))
#15 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\App\Cache\FlushCacheByTags->aroundSave(Object(Magento\Customer\Model\ResourceModel\Customer\Interceptor), Object(Closure), Object(Magento\Customer\Model\Backend\Customer))
#16 /xxx/xxx/xxx/var/generation/Magento/Customer/Model/ResourceModel/Customer/Interceptor.php(117): Magento\Customer\Model\ResourceModel\Customer\Interceptor->___callPlugins('save', Array, Array)
#17 /xxx/xxx/xxx/vendor/magento/framework/Model/AbstractModel.php(631): Magento\Customer\Model\ResourceModel\Customer\Interceptor->save(Object(Magento\Customer\Model\Backend\Customer))
#18 /xxx/xxx/xxx/vendor/magento/module-customer/Model/ResourceModel/CustomerRepository.php(193): Magento\Framework\Model\AbstractModel->save()
#19 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Customer\Model\ResourceModel\CustomerRepository->save(Object(Magento\Customer\Model\Data\Customer), NULL)
#20 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->___callParent('save', Array)
#21 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Array, 'update_newslett...')
#22 /xxx/xxx/xxx/vendor/magento/module-newsletter/Model/Plugin/CustomerPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Customer\Model\Data\Customer), NULL)
#23 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Newsletter\Model\Plugin\CustomerPlugin->aroundSave(Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Object(Closure), Object(Magento\Customer\Model\Data\Customer), NULL)
#24 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Array, 'transactionWrap...')
#25 /xxx/xxx/xxx/vendor/magento/module-customer/Model/Plugin/CustomerRepository/TransactionWrapper.php(44): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Customer\Model\Data\Customer), NULL)
#26 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Customer\Model\Plugin\CustomerRepository\TransactionWrapper->aroundSave(Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Object(Closure), Object(Magento\Customer\Model\Data\Customer))
#27 /xxx/xxx/xxx/var/generation/Magento/Customer/Model/ResourceModel/CustomerRepository/Interceptor.php(26): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->___callPlugins('save', Array, Array)
#28 /xxx/xxx/xxx/vendor/magento/module-customer/Controller/Adminhtml/Index/InlineEdit.php(201): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->save(Object(Magento\Customer\Model\Data\Customer))
#29 /xxx/xxx/xxx/vendor/magento/module-customer/Controller/Adminhtml/Index/InlineEdit.php(113): Magento\Customer\Controller\Adminhtml\Index\InlineEdit->saveCustomer(Object(Magento\Customer\Model\Data\Customer))
#30 /xxx/xxx/xxx/var/generation/Magento/Customer/Controller/Adminhtml/Index/InlineEdit/Interceptor.php(24): Magento\Customer\Controller\Adminhtml\Index\InlineEdit->execute()
#31 /xxx/xxx/xxx/vendor/magento/framework/App/Action/Action.php(102): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->execute()
#32 /xxx/xxx/xxx/vendor/magento/module-backend/App/AbstractAction.php(226): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#33 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#34 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->___callParent('dispatch', Array)
#35 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Array, 'adminAuthentica...')
#36 /xxx/xxx/xxx/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(143): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#37 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#38 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Array, 'adminMassaction...')
#39 /xxx/xxx/xxx/vendor/magento/module-backend/App/Action/Plugin/MassactionKey.php(33): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#40 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Backend\App\Action\Plugin\MassactionKey->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#41 /xxx/xxx/xxx/var/generation/Magento/Customer/Controller/Adminhtml/Index/InlineEdit/Interceptor.php(39): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->___callPlugins('dispatch', Array, Array)
#42 /xxx/xxx/xxx/vendor/magento/framework/App/FrontController.php(55): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#43 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#44 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#45 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#46 /xxx/xxx/xxx/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#47 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#48 /xxx/xxx/xxx/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#49 /xxx/xxx/xxx/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#50 /xxx/xxx/xxx/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#51 /xxx/xxx/xxx/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#52 {main}

Next Zend_Db_Statement_Exception: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'value_id' in 'where clause', query was: DELETE FROM `customer_entity` WHERE (value_id IN ('')) in /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php:235
Stack trace:
#0 /xxx/xxx/xxx/vendor/magento/framework/DB/Statement/Pdo/Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#1 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Statement.php(303): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#2 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#3 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('DELETE FROM `cu...', Array)
#4 /xxx/xxx/xxx/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(465): Zend_Db_Adapter_Pdo_Abstract->query('DELETE FROM `cu...', Array)
#5 /xxx/xxx/xxx/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(520): Magento\Framework\DB\Adapter\Pdo\Mysql->_query('DELETE FROM `cu...', Array)
#6 /xxx/xxx/xxx/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(664): Magento\Framework\DB\Adapter\Pdo\Mysql->query('DELETE FROM `cu...')
#7 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/AbstractEntity.php(1537): Zend_Db_Adapter_Abstract->delete('customer_entity', '(value_id IN ('...')
#8 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/AbstractEntity.php(1455): Magento\Eav\Model\Entity\AbstractEntity->_processAttributeValues()
#9 /xxx/xxx/xxx/vendor/magento/module-eav/Model/Entity/VersionControl/AbstractEntity.php(91): Magento\Eav\Model\Entity\AbstractEntity->_processSaveData(Array)
#10 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Eav\Model\Entity\VersionControl\AbstractEntity->save(Object(Magento\Customer\Model\Backend\Customer))
#11 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Model\ResourceModel\Customer\Interceptor->___callParent('save', Array)
#12 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\Customer\Interceptor), Array, 'clean_cache')
#13 /xxx/xxx/xxx/vendor/magento/framework/App/Cache/FlushCacheByTags.php(71): Magento\Customer\Model\ResourceModel\Customer\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Customer\Model\Backend\Customer))
#14 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\App\Cache\FlushCacheByTags->aroundSave(Object(Magento\Customer\Model\ResourceModel\Customer\Interceptor), Object(Closure), Object(Magento\Customer\Model\Backend\Customer))
#15 /xxx/xxx/xxx/var/generation/Magento/Customer/Model/ResourceModel/Customer/Interceptor.php(117): Magento\Customer\Model\ResourceModel\Customer\Interceptor->___callPlugins('save', Array, Array)
#16 /xxx/xxx/xxx/vendor/magento/framework/Model/AbstractModel.php(631): Magento\Customer\Model\ResourceModel\Customer\Interceptor->save(Object(Magento\Customer\Model\Backend\Customer))
#17 /xxx/xxx/xxx/vendor/magento/module-customer/Model/ResourceModel/CustomerRepository.php(193): Magento\Framework\Model\AbstractModel->save()
#18 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Customer\Model\ResourceModel\CustomerRepository->save(Object(Magento\Customer\Model\Data\Customer), NULL)
#19 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->___callParent('save', Array)
#20 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Array, 'update_newslett...')
#21 /xxx/xxx/xxx/vendor/magento/module-newsletter/Model/Plugin/CustomerPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Customer\Model\Data\Customer), NULL)
#22 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Newsletter\Model\Plugin\CustomerPlugin->aroundSave(Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Object(Closure), Object(Magento\Customer\Model\Data\Customer), NULL)
#23 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'save', Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Array, 'transactionWrap...')
#24 /xxx/xxx/xxx/vendor/magento/module-customer/Model/Plugin/CustomerRepository/TransactionWrapper.php(44): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Customer\Model\Data\Customer), NULL)
#25 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Customer\Model\Plugin\CustomerRepository\TransactionWrapper->aroundSave(Object(Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor), Object(Closure), Object(Magento\Customer\Model\Data\Customer))
#26 /xxx/xxx/xxx/var/generation/Magento/Customer/Model/ResourceModel/CustomerRepository/Interceptor.php(26): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->___callPlugins('save', Array, Array)
#27 /xxx/xxx/xxx/vendor/magento/module-customer/Controller/Adminhtml/Index/InlineEdit.php(201): Magento\Customer\Model\ResourceModel\CustomerRepository\Interceptor->save(Object(Magento\Customer\Model\Data\Customer))
#28 /xxx/xxx/xxx/vendor/magento/module-customer/Controller/Adminhtml/Index/InlineEdit.php(113): Magento\Customer\Controller\Adminhtml\Index\InlineEdit->saveCustomer(Object(Magento\Customer\Model\Data\Customer))
#29 /xxx/xxx/xxx/var/generation/Magento/Customer/Controller/Adminhtml/Index/InlineEdit/Interceptor.php(24): Magento\Customer\Controller\Adminhtml\Index\InlineEdit->execute()
#30 /xxx/xxx/xxx/vendor/magento/framework/App/Action/Action.php(102): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->execute()
#31 /xxx/xxx/xxx/vendor/magento/module-backend/App/AbstractAction.php(226): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#32 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#33 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->___callParent('dispatch', Array)
#34 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Array, 'adminAuthentica...')
#35 /xxx/xxx/xxx/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(143): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#36 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#37 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Array, 'adminMassaction...')
#38 /xxx/xxx/xxx/vendor/magento/module-backend/App/Action/Plugin/MassactionKey.php(33): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#39 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Backend\App\Action\Plugin\MassactionKey->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#40 /xxx/xxx/xxx/var/generation/Magento/Customer/Controller/Adminhtml/Index/InlineEdit/Interceptor.php(39): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->___callPlugins('dispatch', Array, Array)
#41 /xxx/xxx/xxx/vendor/magento/framework/App/FrontController.php(55): Magento\Customer\Controller\Adminhtml\Index\InlineEdit\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#42 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#43 /xxx/xxx/xxx/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#44 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#45 /xxx/xxx/xxx/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#46 /xxx/xxx/xxx/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#47 /xxx/xxx/xxx/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#48 /xxx/xxx/xxx/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#49 /xxx/xxx/xxx/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#50 /xxx/xxx/xxx/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#51 {main} [] []

**Comment by flnative** [comment]
Read this and see if this helps. https://github.com/magento/magento2/issues/7796

**Comment by endergreen** [comment]
@flnative thanks for that thread but that doesn't help either.  It is with any attempt to save the customer that it occurs.  I could have been trying to change the gender or name instead of the customer group with the same result.

**Comment by endergreen** [comment]
I did get the same error when upgrading to 2.1.1 or 2.1.2 from 2.1.0 as well, in stead of just going all the way to 2.1.3.

**Comment by endergreen** [comment]
A registered user also cannot complete the checkout process failing with the same error.  By putting a try-catch around the $this->customerRepository->save($customer); on line 84 of Magento\Quote\Model\CustomrManagement the exception can be caught and logged to see what is occurring.  If that try-catch is not there then there is just an error message to the screen for the user that says "An error occurred on the server.  Please try to place the order again."

**Comment by xmav** [comment]
@endergreen Do you have any custom attributes for Customer?

**Comment by endergreen** [comment]
@xmav, yes, we do.  There are a couple of extensions that have add some to save additional information like profile ids and such for other systems.

**Comment by xmav** [comment]
@endergreen  Unfortunately, I could not reproduce the issue on clean code base(without extensions).
Looks like issue introduced in modules that you use.
Would be great if you can provide steps to reproduce this issue on clean code base.

**Comment by endergreen** [comment]
After you asked about the customer attributes, I started looking at those that were created through the extension setup scripts.  One of those attributes had a backend type of "static" but was not added to the customer entity table as a field.  Once the backend type was changed, the errors went away and the customer was able to be saved through the inline edit as well as being able to place an order for a registered customer.  Looks like a misconfigured customer attribute was the cause but not sure why this would have worked with 2.1.0 and earlier versions.
